### ISSUE TITLE
All features passed displayed before feature results are shown when using tags

### ISSUE THREAD

**Issue by leiyangyou** [meadsteve/white-bread#27]

**Comment by meadsteve** [comment]
@leiyangyou sorry I'm not sure I understand what the issue is. Could you give an example or try rewording the issue?

**Comment by leiyangyou** [comment]
Hi @meadsteve,

here is an example output, initially I thought it was to do with the usage
of tags. looks like that's not the case

mix white_bread.run

```
Navigate to New Post ---> ok
Shows home feed ---> ok
All features passed.
Upvote a post ---> ok
Cancel a post upvote ---> ok
Upvote a downvoted post ---> ok
Downvote a post ---> ok
Cancel a post downvote ---> ok
Downvote a upvoted post ---> ok
```

**Comment by meadsteve** [comment]
@leiyangyou that makes sense. Thanks for clarifying. I think it's because the progress reporter listens to messages and outputs asynchronously yet the feature result is directly output by the main process meaning it can happen before the progress reporter has printed all the other results.

I'll take a look at reworking this as it doesn't make any sense for ```All features passed``` to be printed in the middle.

**Comment by meadsteve** [comment]
@leiyangyou this should be fixed by 2e4fe9b87eba089210f222ee95d91cf66e1a6870

**Comment by leiyangyou** [comment]
Thanks @meadsteve
